src/main/java/com/app/notebook/service/FileService.java

```diff
@@ -0,0 +1,50 @@
+package com.app.notebook.service;
+
+import com.app.notebook.model.File;
+import com.app.notebook.model.dto.UploadFileDto;
+import com.app.notebook.model.mapper.Mapper;
+import com.app.notebook.repository.FileRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.UUID;
+
+@Service
+public class FileService {
+
+    private FileRepository fileRepository;
+    private Mapper mapper;
+
+    @Autowired
+    public FileService(FileRepository fileRepository, Mapper mapper) {
+        this.fileRepository = fileRepository;
+        this.mapper = mapper;
+    }
+
+    public File save(UploadFileDto userFile) throws IOException {
+        java.io.File fileStored = saveToDisk(userFile);
+        File file = mapper.map(userFile);
+        file.setPath(fileStored.getPath());
+
+        return fileRepository.save(file);
+    }
+
+    private java.io.File saveToDisk(UploadFileDto userFile) throws IOException {
+        UUID id = UUID.randomUUID();
+        java.io.File file = new java.io.File(id.toString() + ".txt");
```

Comment:
not cool man, daca urca altceva in alt format s-a stricat treaba... momentan poti pune inca un field pe dto cu extensia pana aflam cum o putem obtine din fisier :P